### PR TITLE
feat(ecocredit/core): UpdateProjectAdmin/Metadata Msg implementations

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds MsgUpdateProjectAdmin 
- adds MsgUpdateProjectMetadata
- implements sdk.Msg for both

ref: #979 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/core/msg_update_project_admin_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"gotest.tools/v3/assert"
+)
+
+func TestMsgUpdateProjectAdmin_ValidateBasic(t *testing.T) {
+	addr := sdk.AccAddress("addr1-----------------").String()
+	addr2 := sdk.AccAddress("addr2-----------------").String()
```
Comment (ryanchristo) [inline x/ecocredit/core/msg_update_project_admin_test.go:13]: A valid address will still be generated without `-----------------`.

```suggestion
	addr := sdk.AccAddress("addr1").String()
	addr2 := sdk.AccAddress("addr2").String()
```

File: x/ecocredit/core/msg_update_project_metadata.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+var _ legacytx.LegacyMsg = &MsgUpdateProjectMetadata{}
+
+func (m MsgUpdateProjectMetadata) ValidateBasic() error {
+	if _, err := types.AccAddressFromBech32(m.Admin); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrap(err.Error())
+	}
+	metadataLen := len(m.NewMetadata)
+	if metadataLen == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("metadata cannot be empty")
+	}
```
Comment (ryanchristo) [inline x/ecocredit/core/msg_update_project_metadata.go:20]: Metadata is currently any arbitrary metadata, including empty metadata. We currently don't check for empty metadata when creating or updating a credit class, when creating a project, or when creating a credit batch. Not sure if there is a use case for removing metadata from a project but I think we should be consistent for now and maybe further discuss on our product call.

File: x/ecocredit/core/msg_update_project_metadata_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/tendermint/libs/rand"
+	"gotest.tools/v3/assert"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+func TestMsgUpdateProjectMetadata_ValidateBasic(t *testing.T) {
+	addr := sdk.AccAddress("addr1---------------------").String()
```
Comment (ryanchristo) [inline x/ecocredit/core/msg_update_project_metadata_test.go:15]:
```suggestion
	addr := sdk.AccAddress("addr1").String()
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. A couple minor suggestions. Pre-approving.

File: x/ecocredit/core/msg_update_project_metadata.go (hunk 2 above)
Comment (technicallyty) [reply x/ecocredit/core/msg_update_project_metadata.go]: https://github.com/regen-network/regen-ledger/pull/1009/commits/0f798b40e62794309e98ed0a5a35658a67519404

fixed

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**